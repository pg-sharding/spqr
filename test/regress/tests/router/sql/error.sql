\c spqr-console
CREATE DISTRIBUTION ds1 COLUMN TYPES integer;

CREATE KEY RANGE kridi1 from 1 route to sh1 FOR DISTRIBUTION ds1;
CREATE KEY RANGE kridi2 from 11 route to sh2 FOR DISTRIBUTION ds1;
ALTER DISTRIBUTION ds1 ATTACH RELATION x DISTRIBUTION KEY id;

\c regress

CREATE TABLE x(id int);

SELECT * FROM x WHERE id = 1;
SELECT * FROM x WHERE ixxxd = 1;
SELECT * FROM x WHERE ixxxd = 1 iuwehiuhweui;
SELECT * FROM x WHERE id = 1;

DROP TABLE x;

\c spqr-console
DROP DISTRIBUTION ALL CASCADE;
DROP KEY RANGE ALL;
