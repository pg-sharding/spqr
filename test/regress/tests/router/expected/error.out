\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

CREATE DISTRIBUTION ds1 COLUMN TYPES integer;
    add distribution    
------------------------
 distribution id -> ds1
(1 row)

CREATE KEY RANGE kridi1 from 1 route to sh1 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 1
(1 row)

CREATE KEY RANGE kridi2 from 11 route to sh2 FOR DISTRIBUTION ds1;
 add key range 
---------------
 bound -> 11
(1 row)

ALTER DISTRIBUTION ds1 ATTACH RELATION x DISTRIBUTION KEY id;
      attach table      
------------------------
 relation name   -> x
 distribution id -> ds1
(2 rows)

\c regress
CREATE TABLE x(id int);
NOTICE: send query to shard(s) : sh1,sh2
SELECT * FROM x WHERE id = 1;
NOTICE: send query to shard(s) : sh1
 id 
----
(0 rows)

SELECT * FROM x WHERE ixxxd = 1;
NOTICE: send query to shard(s) : sh1,sh2
ERROR:  column "ixxxd" does not exist
LINE 1: SELECT * FROM x WHERE ixxxd = 1;
                              ^
SELECT * FROM x WHERE ixxxd = 1 iuwehiuhweui;
ERROR:  client processing error: error processing query 'SELECT * FROM x WHERE ixxxd = 1 iuwehiuhweui;': syntax error on pos 32, tx status IDLE
SELECT * FROM x WHERE id = 1;
NOTICE: send query to shard(s) : sh1
 id 
----
(0 rows)

DROP TABLE x;
NOTICE: send query to shard(s) : sh1,sh2
\c spqr-console

		SPQR router admin console
	Here you can configure your routing rules
------------------------------------------------
	You can find documentation here 
https://github.com/pg-sharding/spqr/tree/master/docs

DROP DISTRIBUTION ALL CASCADE;
   drop distribution    
------------------------
 distribution id -> ds1
(1 row)

DROP KEY RANGE ALL;
 drop key range 
----------------
(0 rows)

